platform/ios/detect.py

```diff
@@ -54,20 +54,15 @@ def configure(env):
 
     ## Build type
 
-    if env["target"].startswith("release"):
+    # TODO: Re-evaluate the need for this / streamline with common config.
+    if not env["target"].endswith("-dev"):
         env.Append(CPPDEFINES=["NDEBUG", ("NS_BLOCK_ASSERTIONS", 1)])
-        if env["optimize"] == "speed":  # optimize for speed (default)
-            # `-O2` is more friendly to debuggers than `-O3`, leading to better crash backtraces
-            # when using `target=release_debug`.
-            opt = "-O3" if env["target"] == "release" else "-O2"
-            env.Append(CCFLAGS=[opt, "-ftree-vectorize", "-fomit-frame-pointer"])
-            env.Append(LINKFLAGS=[opt])
-        elif env["optimize"] == "size":  # optimize for size
-            env.Append(CCFLAGS=["-Os", "-ftree-vectorize"])
-            env.Append(LINKFLAGS=["-Os"])
-
-    elif env["target"] == "debug":
-        env.Append(CCFLAGS=["-gdwarf-2", "-O0"])
+        if env["optimize"].startswith("speed"):
+            env.Append(CCFLAGS=["-ftree-vectorize", "-fomit-frame-pointer"])
+        elif env["optimize"] == "size":
+            env.Append(CCFLAGS=["-ftree-vectorize"])
+    else:
+        env.Append(CCFLAGS=["-gdwarf-2"])
         env.Append(CPPDEFINES=["_DEBUG", ("DEBUG", 1)])
```

Comment:
@bruvzg
- Does it make a difference for iOS if we use `-g` or `-gdwarf-2`? If not we can remove that first line.
- Worth noting that the change will also make iOS `template-dev` build (`optimize=debug`) use `-Og` like other platforms instead of `-O0` like before.
- `_DEBUG` is likely unnecessary as this is a MSVC specific macro.
- Not sure what `DEBUG=1` does and why this is only defined on iOS.

Reply:
Macros are probably just copy-pasted for other platform, I do not think it will do anything.

`gdwarf-2` is likely also unnecessary, probably for using debuggers on old macOS versions (IIRC, default for current clang is drawf-4, and Xcode on macOS older than 10.10 do not support it).

Reply:
Changed it in #66303.